Bottom: 8ceb90b5863e37f3cca521d81fbc67011be1eb3a
Top:    7e66daaa82f35a66fc8df9b5f41719e43fea139a
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-03-05 16:23:31 +0000

polygon1.c: Fix unitialised output polygon upon when hitting allocation failure

This was a bug present in the poly_Boolean() routine. We need to set the output
polygon to NULL if we are going to return early on memory failure. This was
noticed during general review after observing problems due to our copy routines
incorrectly returning FALSE when copying a NULL polygon.


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 33a2212..511bd21 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2326,6 +2326,8 @@ poly_Boolean (const POLYAREA * a_org, const POLYAREA * b_org,
 {
   POLYAREA *a = NULL, *b = NULL;
 
+  *res = NULL;
+
   if (!poly_M_Copy0 (&a, a_org) || !poly_M_Copy0 (&b, b_org))
     return err_no_memory;
